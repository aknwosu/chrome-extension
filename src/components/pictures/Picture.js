import React from 'react'
import styled from 'styled-components'

import heartOutlineIcon from '../../ui/heart-outline.svg'

import heartIcon from '../../ui/heart-icon.svg'

export const Picture = (props) => {
  const { picture, favoritesIDs, setFavorite } = props
  const isFavorite = favoritesIDs.includes(picture.id)
  const title = picture.title.length > 25 ? `${picture.title.slice(0, 23)}...` : picture.title
  return (
    <PictureWrapper>
      <div>{title}</div>
      <StyledImage src={picture.thumbnailUrl}>
        <StyledFaveCta isFavorite={isFavorite} onClick={() => setFavorite(picture.id)} />
      </StyledImage>
    </PictureWrapper>
  )
}

export default Picture


export const StyledImage = styled.div`
  height: 150px;
  background: ${({ src }) => `url(${src})`};
  position: relative;
  background-position: center;
  width: 150px;
  background-repeat: no-repeat;
  margin: auto;
`
const PictureWrapper = styled.div`
  height: 180px;
  margin: auto;
  margin-bottom: 6px;
`
export const StyledFaveCta = styled.div`
  position: absolute;
  top: 10px;
  right: 10px;
  width: 28px;
  height: 28px;
	background-image: url("${({ isFavorite }) => (isFavorite ? heartIcon : heartOutlineIcon)}");
	background-position: top right;
	background-repeat: no-repeat;
	background-size: 28px;
	width: 28px;
	height: 28px;
  cursor: pointer;
`