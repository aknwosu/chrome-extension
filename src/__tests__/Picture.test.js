import React from 'react';
import { shallow, mount } from 'enzyme'
import { Picture, StyledFaveCta } from '../components/Pictures/Picture'
import { create } from 'react-test-renderer'


const picture = {
  "albumId": 1,
  "id": 2,
  "title": "reprehenderit est deserunt velit ipsam",
  "url": "https://via.placeholder.com/600/771796",
  "thumbnailUrl": "https://via.placeholder.com/150/771796"
}

describe('Picture', () => {
  let wrapper;
  const favoritesIDs = [2]
  const setFavorite = jest.fn()
  const comp = <Picture picture={picture} favoritesIDs={favoritesIDs} setFavorite={setFavorite} />
  beforeEach(() => {
    wrapper = shallow(comp).dive()
  })
  it('should match snapshot', () => {
    const c = create(comp);
    expect(c.toJSON()).toMatchSnapshot();
  })

  it('should call setFavorite on click of CTA', () => {
    wrapper.find(StyledFaveCta).simulate('click');
    expect(setFavorite).toHaveBeenCalled();
  })

  it('should chek if the picture ia a favorite', () => {  
    let favorited = wrapper.find(StyledFaveCta).props().isFavorite
    expect(favorited).toBe(true);
  })

  it('should call check is the picture has not been favorited', () => {
    let newWrapper = mount(<Picture picture={picture} favoritesIDs={[]} setFavorite={setFavorite} />)
    let favorited = newWrapper.find(StyledFaveCta).props().isFavorite
    expect(favorited).toBe(false);
  })
})