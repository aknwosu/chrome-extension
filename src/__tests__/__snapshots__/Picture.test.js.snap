// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picture should match snapshot 1`] = `
<div
  className="sc-bwzfXH bTEMlB"
>
  <div>
    reprehenderit est deser...
  </div>
  <div
    className="sc-bdVaJa YDxLb"
    src="https://via.placeholder.com/150/771796"
  >
    <div
      className="sc-htpNat cRXqux"
      onClick={[Function]}
    />
  </div>
</div>
`;
